dali/operators/decoder/nvjpeg/decoupled_api/nvjpeg_decoder_decoupled_api.h

```diff
@@ -35,6 +35,84 @@
 
 namespace dali {
 
+namespace memory {
+
+using Buffer = std::pair<void*, size_t>;
+using BufferPool = std::map<std::thread::id, std::vector<Buffer>>;
+static BufferPool device_buffer_pool_;
+static BufferPool pinned_buffer_pool_;
+
+static int DeviceNew(void **ptr, size_t size) {
+  std::thread::id this_id = std::this_thread::get_id();
+  std::cout << "Requesting device memory for Thread " << this_id << ": " << size << " bytes, ptr = ";
+  auto it = device_buffer_pool_.find(this_id);
```

Comment:
I'm afraid you need to add a lock here, as you may have two pipelines created one after another. And one would use this allocator and the second would try to allocate ahead something inside so concurrent access and modification of `device_buffer_pool_` may happen.

Reply:
True that, done